tests/thread_float/tests/01-run.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+#  Copyright (C) 2021 Otto-von-Guericke-Universität Magdeburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+# @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+
+import sys
+from testrunner import run
+
+
+def assertAlmostEqual(first, second, delta=0.05):
+    assert first + delta > second
+    assert first - delta < second
+
+
+def testfunc(child):
+    child.expect_exact("THREADS CREATED")
+    child.expect_exact("THREAD t1 start")
+    child.expect_exact("THREAD t2 start")
+    child.expect_exact("THREAD t3 start")
+
+    child.expect(r"t(\d): (\d{3}\.\d+)\r\n")
+    first_thread = int(child.match.group(1))
+    # Note: intentionally keeping the float output as string to also test that printf("%f", ...)
+    # prints the exact same char sequence for the same float value each time
+    first_result = child.match.group(2)
+    child.expect(r"t(\d): (\d{3}\.\d+)")
```

Comment:
```suggestion
    child.expect(r"t(\d): (\d{3}\.\d+)\r\n")
```

Just noticed that this is documented here: https://github.com/RIOT-OS/RIOT/tree/master/tests#automated-tests-guidelines

There are other candidates for this below.

Reply:
> Just noticed that this is documented here: https://github.com/RIOT-OS/RIOT/tree/master/tests#automated-tests-guidelines

Thanks. Was too lazy to search it for https://github.com/RIOT-OS/RIOT/pull/16901#issuecomment-930280528 ^^"